### PR TITLE
Add zero-storage

### PR DESCRIPTION
**Pull request by trdarr**
As seen on ~~TV~~ [toopher/toopher-perl](https://github.com/toopher/toopher-perl)!

### CODE REVIEW

**Comment by egrim** [comment]
Looks good to me - port over naming changes as discussed in cave and we're good to go.

**Comment by smholloway** [comment]
:shipit: :baby:

File: README.md [hunk 1]
```diff
@@ -44,6 +44,37 @@ if (auth_status.pending == false and auth_status.granted == true):
...
+except UserDisabledError:
+    # you have marked this user as disabled in the Toopher API.
+except UnknownUserError:
+    # This user has not yet paired a mobile device with their account.  Pair them
+    # using api.pair() as described above, then re-try authentication
+except UnknownTerminalError:
+    # This user has not assigned a "Friendly Name" to this terminal identifier.
+    # Prompt them to enter a terminal name, then submit that "friendly name" to
+    # the Toopher API:
+    #   api.assign_user_friendly_name_to_terminal(user_name, terminal_friendly_name, terminal_identifier)
```
Comment (smholloway) [inline README.md:70]: This method is now called `create_user_terminal`

File: tests.py [hunk 2]
```diff
@@ -177,6 +177,56 @@ def test_access_arbitrary_keys_in_authentication_status(self):
...
+    def test_disabled_pairing_raises_correct_error(self):
+        api = toopher.ToopherApi('key', 'secret', api_url='https://toopher.test/v1')
+        api.client = HttpClientMock({
+            'https://toopher.test/v1/authentication_requests/initiate':
+                ({'status': 409}, json.dumps(
+                    {'error_code': 601,
+                     'error_message': 'pairing has not been authorized'}))})
+        with self.assertRaises(toopher.PairingDeactivatedError):
+            auth_request = api.authenticate_by_user_name('user', 'terminal name')
+
```
Comment (smholloway) [inline tests.py:229]: Can you test the `set_enable_toopher_for_user` method too? I think there should be three cases: 
- [ ] no user found -> error
- [ ] multiple users found -> error
- [ ] one user found -> success

File: toopher/__init__.py [hunk 3]
```diff
@@ -6,6 +6,14 @@
 DEFAULT_BASE_URL = "https://api.toopher.com/v1"
 VERSION = "1.0.6"
```
Comment (smholloway) [inline toopher/__init__.py:7]: Psssst, bump up the VERSION to `1.1.0`, please.

File: toopher/__init__.py [hunk 4]
```diff
@@ -60,30 +68,62 @@ def get_authentication_status(self, authentication_request_id):
...
+        response, content = self.client.request(uri, method, data, headers=header_data)
         try:
-            result = json.loads(content)
-        except Exception, e:
-            raise ToopherApiError("Response from server could not be decoded as JSON: %s" % e)
+            content = json.loads(content)
+        except ValueError:
+            raise ToopherApiError('Response from server could not be decoded as JSON.')
+
+        if int(response['status']) > 300:
```
Comment (smholloway) [inline toopher/__init__.py:109]: Is `300` the right comparison? `300`s are redirects; `400`s are client errors; `500`s are server errors. As you said, we're probably more concerned with errors, so we should check `>= 400`.

File: toopher/__init__.py [hunk 5]
```diff
@@ -60,30 +68,62 @@ def get_authentication_status(self, authentication_request_id):
...
+        return self.authenticate('', '', action_name, **kwargs)
+
+    def create_user_terminal(self, user_name, terminal_name, requester_terminal_id):
+        uri = self.base_url + '/user_terminals/create'
+        params = {'user_name': user_name,
+                  'name': terminal_name,
+                  'name_extra': requester_terminal_id}
+        result = self._request(uri, 'POST', params)
+
+    def set_enable_toopher_for_user(self, user_name, enabled):
```
Comment (smholloway) [inline toopher/__init__.py:87]: The [Toopher Perl library](https://github.com/toopher/toopher-perl/blob/master/lib/ToopherAPI.pm) called this method `set_toopher_enabled_for_user`.

**Pull request merged**